core/trino-main/src/main/java/io/trino/metadata/FunctionResolver.java

```diff
@@ -155,6 +183,35 @@ FunctionBinding resolveFunction(Session session, Collection<FunctionMetadata> al
         throw new TrinoException(FUNCTION_NOT_FOUND, message);
     }
 
+    private static List<CatalogSchemaFunctionName> toPath(Session session, QualifiedName name)
+    {
+        List<String> parts = name.getParts();
+        checkArgument(parts.size() <= 3, "Function name can only have 3 parts: " + name);
+        if (parts.size() == 3) {
+            return ImmutableList.of(new CatalogSchemaFunctionName(parts.get(0), parts.get(1), parts.get(2)));
+        }
+
+        if (parts.size() == 2) {
+            String currentCatalog = session.getCatalog()
+                    .orElseThrow(() -> new IllegalArgumentException("Session default catalog must be set to resolve a partial function name: " + name));
+            return ImmutableList.of(new CatalogSchemaFunctionName(currentCatalog, parts.get(0), parts.get(1)));
+        }
+
+        ImmutableList.Builder<CatalogSchemaFunctionName> names = ImmutableList.builder();
+        String functionName = parts.get(0);
+
+        // global namespace
+        names.add(new CatalogSchemaFunctionName(GLOBAL_CATALOG, GLOBAL_SCHEMA, functionName));
+
+        // add resolved path items
+        for (SqlPathElement sqlPathElement : session.getPath().getParsedPath()) {
+            String catalog = sqlPathElement.getCatalog().map(Identifier::getValue).or(session::getCatalog)
+                    .orElseThrow(() -> new IllegalArgumentException("Session default catalog must be set to resolve a partial function name: " + null));
+            names.add(new CatalogSchemaFunctionName(catalog, sqlPathElement.getSchema().getValue(), functionName));
```

Comment:
Could you use `getCanonicalValue()` instead of `getValue()`?
I think it's no difference at the moment, as all the names are eventually lowercased, but it would simplify the future transition to SQL identifier semantics.

Reply:
Sure.  Didn't know that method existed.